### ISSUE TITLE
MIssing permission on Role to get leases

### ISSUE THREAD

**Issue by nsurleraux-railnova** [alexellis/registry-creds#42]
**Describe the bug**
When upgrading `registry-creds` using kubectl, the controller throws an error:
```
E0125 08:06:38.503340       1 leaderelection.go:330] error retrieving resource lock registry-creds-system/8bdecb1a.alexellis.io: leases.coordination.k8s.io "8bdecb1a.alexellis.io" is forbidden: User "system:serviceaccount:registry-creds-system:default" cannot get resource "leases" in API group "coordination.k8s.io" in the namespace "registry-creds-system"
``` 

**To Reproduce**
1. `kubectl apply -f https://raw.githubusercontent.com/alexellis/registry-creds/master/manifest.yaml`
2. Get the log from the controller

**Expected behavior**
The error should not happen and the leases should be updated.

**Are you a GitHub Sponsor (Yes/No?)**

Issues from sponsors get prioritised and a quicker response.

Check at: https://github.com/sponsors/alexellis
- [ ] Yes
- [x] No

**Screenshots or console output**
```
E0125 08:06:38.503340       1 leaderelection.go:330] error retrieving resource lock registry-creds-system/8bdecb1a.alexellis.io: leases.coordination.k8s.io "8bdecb1a.alexellis.io" is forbidden: User "system:serviceaccount:registry-creds-system:default" cannot get resource "leases" in API group "coordination.k8s.io" in the namespace "registry-creds-system"
```

**Operating system and version:**
EKS running Kubernetes 1.21

**List all possible solutions, and your suggested option**
Add the required rule to the `Role`:
```yaml
  - verbs:
      - get
      - create
      - update
      - patch
    apiGroups:
      - coordination.k8s.io
    resources:
      - leases
```

**Additional context**
N/A

**Comment by alexellis** [comment]
Hey thanks for your interest in this project.

Tested PRs are always welcome, would you prefer to do that or to set up [a sponsorship](https://github.com/sponsors/alexellis) and I'll carve out some time for this?

Alex

**Comment by nsurleraux-railnova** [comment]
Hi @alexellis,

I opened a pull request. Let me know if you need more details.

Thanks,

Nicolas

**Comment by nsurleraux-railnova** [comment]
Closed by #43
